### PR TITLE
Add Figma to BabylonJS to Community Extensions

### PR DESCRIPTION
**Pull request by jamessimo**
Was asked by @PirateJC to add my Figma Plugin to the Community Extensions. 

First PR so please feel free to correct me if I have made any mistakes. I have tested the site on local and everything is working as intended.

### CODE REVIEW

**Review by RaananW — APPROVED** [review]
Love it! will let @PatrickRyanMS take a quick look and merge it.
Thanks!!

File: content/communityExtensions/figmaToBabylonJS.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+---
+title: Figma to BabylonJS
+image: /img/extensions/FigmaToBabylonJS.png
+description: FigmaToBabylonJS is a Figma plugin which allows you to export  Figma designs to BabylonJS GUI .json files.
```
Comment (phildremi) [inline content/communityExtensions/figmaToBabylonJS.md]: Extra space here: ``export  Figma``

File: content/communityExtensions/figmaToBabylonJS.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+![Screenshot showing the figma right click menu on the Plugins option](/img/extensions/FigmaToBabylonJS-Step2.png)
+
+It's also good practice to load the UI in gui.babylonjs.com or in your project but loading in the inspector so you can edit the UI after the fact. Just to make sure that the exporting has went the way you expected and there weren't any strange inconsistencies with the design [see inconsistencies and polyfills](/communityExtensions/figmaToBabylonJS#inconstancies--polyfills)
+
+## How it Works
+
+The plugin recursively loops through the selected Figma frame and pulls out all the "nodes" in the design. It will take each node and find a correct BabylonJS GUI object to map to it.
+All the new BabylonJS GUI mapped objects are then added to a empty root container object as children. When the plugin is finished adding all these children, it simply outputs it as a JSON string and writes it to a file and makes it available to download or “stream” from Figma directly.
+
+## Inconstancies & Polyfills
```
Comment (phildremi) [inline content/communityExtensions/figmaToBabylonJS.md]: ``Inconstancies`` should (presumably) be ``Inconsistencies``?

File: content/communityExtensions/figmaToBabylonJS.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+Figma has features that beyond the scope of BabylonJS GUI system. Below is a list of some of the areas which the plug-in has to make a determination on how to interpret things that Figma is asking for that BabylonJS GUI cannot replicate 1:1.
+
+### Handling Images
+
+Images can’t be represented as binary or BASE64 inside the json file so currently the plugin will just set the image src too `source=“./imageControlDefault.jpg”`. After exporting the json its recommended you export your images separately and correct the source addresses either in the JSON file or in code.
```
Comment (phildremi) [inline content/communityExtensions/figmaToBabylonJS.md]: Typos:

*  ``set the image src too`` should be ``set the image src to``?
* ``its recommended`` should be ``it's recommended``?

File: content/communityExtensions/figmaToBabylonJS.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+### Misc Inconstancies
```
Comment (phildremi) [inline content/communityExtensions/figmaToBabylonJS.md]: Seemingly referring to ``Inconsistencies`` again?

File: content/communityExtensions/figmaToBabylonJS.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+On the rightclick menu, select Plugins > Saved Plugins > FigmaToBabylonJS
+![Screenshot showing the figma right click menu on the Plugins option](/img/extensions/FigmaToBabylonJS-Step1.png)
+
+Choose if you want the output to include the background color (changes the "root" background color to the frames fill color)
+Click the “Convert to BabylonJS JSON” button and after a short wait you will see your file explorer. Choose where you want the file to go. For best results, place it directly into your project with the correct file name your code uses, this way the UI will automatically update.
+![Screenshot showing the figma right click menu on the Plugins option](/img/extensions/FigmaToBabylonJS-Step2.png)
+
+It's also good practice to load the UI in gui.babylonjs.com or in your project but loading in the inspector so you can edit the UI after the fact. Just to make sure that the exporting has went the way you expected and there weren't any strange inconsistencies with the design [see inconsistencies and polyfills](/communityExtensions/figmaToBabylonJS#inconstancies--polyfills)
```
Comment (PatrickRyanMS) [inline content/communityExtensions/figmaToBabylonJS.md]: "It's also good practice to load the UI in gui.babylonjs.com or in your project but loading in the inspector so you can edit the UI after the fact" 
This sentence is a little confusing. You are saying it's good practice to load the UI into gui.babylonjs.com, but the second part of the sentence is not clear "or in your project but loading in the inspector".

I think you are saying you "should preview the UI by loading it into either gui.babylonjs.com or your local project using the inspector" but the original wording is a bit confusing

File: content/communityExtensions/figmaToBabylonJS.md (hunk 2 above)
Comment (PatrickRyanMS) [reply content/communityExtensions/figmaToBabylonJS.md]: It seems this word should be "Inconsistencies" or "Unsupported Features"

File: content/communityExtensions/figmaToBabylonJS.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+## How it Works
+
+The plugin recursively loops through the selected Figma frame and pulls out all the "nodes" in the design. It will take each node and find a correct BabylonJS GUI object to map to it.
+All the new BabylonJS GUI mapped objects are then added to a empty root container object as children. When the plugin is finished adding all these children, it simply outputs it as a JSON string and writes it to a file and makes it available to download or “stream” from Figma directly.
```
Comment (PatrickRyanMS) [inline content/communityExtensions/figmaToBabylonJS.md]: There is a concept of "streaming" a UI from Figma directly. It implies that as you make changes to the UI in Figma that the UI in your project will update automatically. Is this just an automated export of the json file or is this actively reloading the Babylon scene when changes to the exported json are detected? I think it's worth another sentence or two to describe what is meant by "streaming" for people who may not be familiar with Figma.

File: content/communityExtensions/figmaToBabylonJS.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+## Inconstancies & Polyfills
+
+Figma has features that beyond the scope of BabylonJS GUI system. Below is a list of some of the areas which the plug-in has to make a determination on how to interpret things that Figma is asking for that BabylonJS GUI cannot replicate 1:1.
```
Comment (PatrickRyanMS) [inline content/communityExtensions/figmaToBabylonJS.md]: "Figma has features that beyond the scope of BabylonJS GUI system." Missing word "are." The sentence should read:

"Figma has features that **are** beyond the scope of BabylonJS GUI system."

File: content/communityExtensions/figmaToBabylonJS.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+Images can’t be represented as binary or BASE64 inside the json file so currently the plugin will just set the image src too `source=“./imageControlDefault.jpg”`. After exporting the json its recommended you export your images separately and correct the source addresses either in the JSON file or in code.
+
+### Misc Inconstancies
+
+- Borders
+  - All borders are treated as "Outer" when porting to BabylonJS. 
+- Colors not 1:1
+  - By default Figma uses an "unmanaged" color space, to make Figma closer represent what will actually be outputted when viewed on the web browser, go to Figma preferences > ColorSpace and set the ColorSpace to sRGB.
+- Polygons
+  - Polygons are not supported by BabylonJS GUI, the plug-in will simply skip over polygons when encountered. If you need to use complex shapes and polygons like this it's recommended that you export it as its own PNG and imported into the BabylonJS project as an image.
```
Comment (PatrickRyanMS) [inline content/communityExtensions/figmaToBabylonJS.md]: "...it's recommended that you export it as its own PNG and imported into the BabylonJS project as an image."

The word "imported" is incorrect here. This part of the sentence should read:

"...it's recommended that you export it as its own PNG and **import it** into the BabylonJS project as an image."

File: content/communityExtensions/figmaToBabylonJS.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+- Lines
+  - Lines are supported by BabylonJS but Figma represents them as an array of vertexes. Right now the plugin will skip over lines but this feature will be added in future.
```
Comment (PatrickRyanMS) [inline content/communityExtensions/figmaToBabylonJS.md]: I believe "vertices" is the more common plural for "vertex" over and also aligns more with the style in the rest of the documentation.

File: content/communityExtensions/figmaToBabylonJS.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+- Gradient Fills
+  - This feature is being worked on, currently the plug-in will simply skip over any gradient used, this will be supported in future.
+- Blending/Multi Fills
+  - Figma allows you to layer fills one on top of the other and use special blending options. Right now the plugin will only use the top most fill. 
+  - Do not use fills that have its visibility turned off. There have been occasions where Figma returns false positives.
```
Comment (PatrickRyanMS) [inline content/communityExtensions/figmaToBabylonJS.md]: There is an inconsistency using the plural "fills" and the singular "its." The correction here would be either:

"Do not use a fill that has its visibility turned off."

or

"Do not use fills that have their visibility turned off."

File: content/communityExtensions/figmaToBabylonJS.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+
+## Help & Support
+
+- If the file output name is “undefined” that means that you are selecting a single item on the design and not the whole frame/artboard.
+- Very large designs may cause the plugin to "hang", please be paitient, if the plugin takes too much time Figma will abort it for you. If you have this issue make the design frame less complex or simplify Components which reference other Componenets.
```
Comment (PatrickRyanMS) [inline content/communityExtensions/figmaToBabylonJS.md]: The first sentence is a run-on and can be broken into three. It's much easier to read:

"Very large designs may cause the plugin to "hang." Please be patient. If the plugin takes too much time, Figma will abort it for you."

The last word on this line is misspelled. "Componenets" should be "Components."

**Review by PatrickRyanMS — CHANGES_REQUESTED** [review]
There are a few grammatical changes and spelling errors to correct before merging.

File: content/communityExtensions/figmaToBabylonJS.md (hunk 2 above)
Comment (jamessimo) [reply content/communityExtensions/figmaToBabylonJS.md]: Thanks for catching this

File: content/communityExtensions/figmaToBabylonJS.md (hunk 5 above)
Comment (jamessimo) [reply content/communityExtensions/figmaToBabylonJS.md]: Agreed, fixed in my latest commit.

File: content/communityExtensions/figmaToBabylonJS.md (hunk 6 above)
Comment (jamessimo) [reply content/communityExtensions/figmaToBabylonJS.md]: Thanks, I removed the streaming thing as its not important. It was more to do with how the file is made but its not important for the end user.

File: content/communityExtensions/figmaToBabylonJS.md (hunk 10 above)
Comment (jamessimo) [reply content/communityExtensions/figmaToBabylonJS.md]: Went with the second option as you can have many fills.

File: content/communityExtensions/figmaToBabylonJS.md (hunk 11 above)
Comment (jamessimo) [reply content/communityExtensions/figmaToBabylonJS.md]: Agreed, fixed it up in my latest commit.

**Comment by jamessimo** [comment]
Are there any other issues? I resolved all the open change requests.

**Review by PatrickRyanMS — APPROVED** [review]

**Comment by PatrickRyanMS** [comment]
@RaananW, it looks like all of the feedback has been addressed. I will let you proceed with final approvals.

**Pull request merged**